src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureKeyVaultReference/AzureKeyVaultSecretProvider.cs

```diff
@@ -14,32 +13,28 @@ namespace Microsoft.Extensions.Configuration.AzureAppConfiguration.AzureKeyVault
     internal class AzureKeyVaultSecretProvider
     {
         private readonly IDictionary<string, SecretClient> _secretClients;
-        private readonly TokenCredential _credential;
-        private readonly Func<Uri, ValueTask<string>> _secretResolver;
+        private Dictionary<string, CachedKeyVaultSecret> _cachedKeyVaultSecrets = new Dictionary<string, CachedKeyVaultSecret>();
```

Comment:
>new Dictionary [](http://example.com/codeflow?start=82&length=14)

We're initializing _secretClients in the constructor. We probably want to follow the same pattern for both.